Offroad-iOS/Offroad-iOS/Global/Components/ORBSuggestionGradient/ORBRecommendationGradientStyle.swift

```diff
@@ -0,0 +1,97 @@
+    
+    private var orbRecommendationBaseBackgroundView: ORBRecommendationBaseBackground? {
+        get { objc_getAssociatedObject(self, &orbRecommendationGradientBaseViewKey) as? ORBRecommendationBaseBackground }
+        set { objc_setAssociatedObject(self, &orbRecommendationGradientBaseViewKey, newValue, .OBJC_ASSOCIATION_RETAIN_NONATOMIC) }
+    }
+    
+    private var isGradientStyleApplied: Bool {
+        get { (objc_getAssociatedObject(self, &isGradientStyleAppliedKey) as? Bool) ?? false }
+        set { objc_setAssociatedObject(self, &isGradientStyleAppliedKey, newValue, .OBJC_ASSOCIATION_RETAIN_NONATOMIC) }
+    }
+    
+    /// 오브의 추천소 그라데이션 적용 시 `layer.border`를 숨겨야 하므로 `borderWidth`를 0으로 설정함.
+    /// 그라데이션을 지울 때 다시 원래의 `borderWidth`로 복원해 주어야 하므로 이를 저장하기 위한 속성.
+    private var borderWidthCache: CGFloat {
+        get { objc_getAssociatedObject(self, &borderWidthCacheKey) as? CGFloat ?? 0 }
+        set { objc_setAssociatedObject(self, &borderWidthCacheKey, newValue, .OBJC_ASSOCIATION_RETAIN_NONATOMIC) }
+    }
+    
+    /// 그라데이션 효과가 적용될 테두리 두께.
+    var gradientBorderWidth: CGFloat {
+        get { return orbRecommendationBaseBackgroundView?.gradientBorderWidth ?? 1 }
+        set { orbRecommendationBaseBackgroundView?.gradientBorderWidth = newValue }
+    }
+    
+    /// 배경에 (블러가 적용된) 그라데이션 효과를 적용할 지 여부. `true`인 경우 배경에도 연한 블러 그라데이션 효과가 적용됨. `false`인 경우 배경은 흰색 적용.
+    var isBackgroundBlurred: Bool {
+        get { return orbRecommendationBaseBackgroundView?.isBackgroundBlurred ?? true }
+        set { orbRecommendationBaseBackgroundView?.isBackgroundBlurred = newValue }
+    }
+    
+    func applyGradientStyle(borderWidth: CGFloat = 1, isBackgroundBlurred: Bool = true) {
+        guard !isGradientStyleApplied else { return }
+        
+        if let baseBackgroundView = orbRecommendationBaseBackgroundView {
+            baseBackgroundView.gradientBorderWidth = borderWidth
+            baseBackgroundView.isBackgroundBlurred = isBackgroundBlurred
+            insertSubview(baseBackgroundView, at: 0)
+        } else {
+            orbRecommendationBaseBackgroundView = ORBRecommendationBaseBackground(
+                borderWidth: borderWidth,
+                isBackgroundBlurred: isBackgroundBlurred
+            )
+            insertSubview(orbRecommendationBaseBackgroundView!, at: 0)
+        }
+        orbRecommendationBaseBackgroundView!.snp.makeConstraints { make in
+            make.edges.equalToSuperview()//.priority(.init(999))
+        }
+        layoutIfNeeded()
+        orbRecommendationBaseBackgroundView!.clipsToBounds = true
+        orbRecommendationBaseBackgroundView!.alpha = 1
+        updateBaseBackgroundCorner()
+        borderWidthCache = layer.borderWidth
+        layer.borderWidth = 0
+        
+        isGradientStyleApplied = true
+    }
+    
+    func removeGradientStyle() {
+        guard isGradientStyleApplied else { return }
+        orbRecommendationBaseBackgroundView?.alpha = 0
```

Comment:
P4

remove가 목적이라면 단순히 alpha 값을 0으로 설정하기보다 아래와 같이 뷰를 superview에서 제거하고 참조를 해제하는 방식이 더 좋지 않을까하는 개인적인 의견입니다.
```suggestion
        orbRecommendationBaseBackgroundView?.removeFromSuperview()
        orbRecommendationBaseBackgroundView = nil
```

Reply:
저도 처음에 아예 뷰를 지우고 다시 생성하는 점을 고려했었는데, 
인스턴스를 완전히 메모리에서 내리고(deallocate) 할당(allocate)하는 과정이 반복되는 것보다 한 번 할당하고 이를 참조하는 게 조금이나마 성능에 도움이 될 것 같다고 생각했습니다.

또, alpha 값을 수정하게 되면 그라데이션을 나타내고 없애는 과정에서 애니메이션을 적용할 수 있기도 합니다.
(사실 지금은 이런 상황이 존재하지 않아서 큰 의미 없긴 합니다..!)

사실
- 지금 상황에서는 그라데이션을 켜고 끌 일이 없음
- 단순 UIView 하나만 생성하고 없애는 것 정도는 성능에 크게 차이 없을 듯
등의 이유로 인해 큰 차이는 없을 것 같다고 생각합니다!

alpha 값을 수정하는 경우의 단점을 생각해보자면...
만약 많은 뷰에서 그라데이션을 켠 후, 그 많은 뷰에서 다시 그라데이션을 끈다면 그라데이션 배경 뷰가 메모리에서 해제되지 않고 남아있으니 불필요하게 메모리를 잡고 있는 경우가 발생할 수 있긴 합니다만, 이런 경우도 거의 없을 것 같아서...

나중에 필요한 경우 상황에 맞게 수정하면 될 것 같습니다!

질문 감사합니다!